ui/components/config/provider-form.tsx

```
@@ -0,0 +1,834 @@
+    field: keyof KeyType,
+    value: string | number | string[]
+  ) => {
+    const newKeys = [...keys];
+    const keyToUpdate = { ...newKeys[index] };
+
+    if (field === "models" && Array.isArray(value)) {
+      keyToUpdate.models = value;
+    } else if (field === "value" && typeof value === "string") {
+      keyToUpdate.value = value;
+    } else if (field === "weight" && typeof value === "string") {
+      keyToUpdate.weight = parseFloat(value) || 1.0;
```

Biome (1.9.4)

[error] Use Number.parseFloat instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.parseFloat instead.


(lint/style/useNumberNamespace)